src/PowerShell/Microsoft.WinGet.Configuration.Engine/Commands/ConfigurationCommand.cs

```diff
@@ -13,74 +13,278 @@ namespace Microsoft.WinGet.Configuration.Engine.Commands
+            // Start task.
+            var runningTask = this.RunOnMTA<PSConfigurationSet>(
+                async () => await this.OpenConfigurationSetAsync(configFile, executionPolicy));
+
+            this.Wait(runningTask);
+            this.WriteObject(runningTask.Result);
         }
 
         /// <summary>
-        /// Invoke configuration. Waits until completed.
+        /// Gets the details of a configuration set.
         /// </summary>
-        public void Invoke()
+        /// <param name="psConfigurationSet">PSConfigurationSet.</param>
+        public void GetDetails(PSConfigurationSet psConfigurationSet)
         {
-            // Start task.
-            var runningTask = this.RunOnMTA(this.InvokeAsync);
+            if (!psConfigurationSet.HasDetails)
+            {
+                if (!psConfigurationSet.CanProcess())
+                {
+                    // TODO: better exception or just write info and return null.
+                    throw new Exception("Someone is using me!!!");
+                }
+
+                var runningTask = this.RunOnMTA<PSConfigurationSet>(
+                async () =>
+                {
+                    try
+                    {
+                        psConfigurationSet = await this.GetSetDetailsAsync(psConfigurationSet);
+                    }
+                    finally
+                    {
+                        psConfigurationSet.DoneProcessing();
+                    }
+
+                    return psConfigurationSet;
+                });
+
+                this.Wait(runningTask);
+                psConfigurationSet = runningTask.Result;
+            }
+            else
+            {
+                this.WriteWarning("Details already obtained for this set");
+            }
 
-            // Wait for it to complete or being cancelled.
+            this.WriteObject(psConfigurationSet);
+        }
+
+        /// <summary>
+        /// Starts configuration.
+        /// </summary>
+        /// <param name="psConfigurationSet">PSConfigurationSet.</param>
+        public void StartApply(PSConfigurationSet psConfigurationSet)
+        {
+            if (psConfigurationSet.Set.State == ConfigurationSetState.Completed)
+            {
+                this.WriteWarning("Processing this set is completed");
+                return;
+            }
+
+            if (!psConfigurationSet.CanProcess())
+            {
+                // TODO: better exception or just write info and return null.
+                throw new Exception("Someone is using me!!!");
+            }
+
+            var runningTask = this.StartApplyInternal(psConfigurationSet);
+            this.WriteObject(new PSConfigurationTask(runningTask, this));
+        }
+
+        /// <summary>
+        /// Applies configuration.
+        /// </summary>
+        /// <param name="psConfigurationSet">PSConfigurationSet.</param>
+        public void Apply(PSConfigurationSet psConfigurationSet)

```

Comment:
You should create another helper(s) so that this method can always be as simple as `ContinueHelper(StartApplyHelper(set));`.